src/streams.rs

```diff
@@ -70,7 +70,7 @@ impl<'a> StreamAdapter<'a> {
 impl<'a> From<&'a dyn Stream> for StreamAdapter<'a> {
     #[allow(invalid_reference_casting)]
     fn from(stream: &'a dyn Stream) -> Self {
-        let stream = &*stream as *const dyn Stream as *mut dyn Stream;
+        let stream = stream as *const dyn Stream as *mut dyn Stream;
```

Comment:
The old version and new version are the same type, as pointed out by clippy. Lemme know if I'm missing something, @gpeacock. I know the binding stuff can be crazy. Maybe the `&*` is required anyway?

Reply:
I'll trust clippy on this one. That's a remnant of older code.